syntax = "proto3";

package masterworker;

service MasterWorker {
  rpc mapReduce(MasterQuery) returns (WorkerReply) {}
}

message TempFiles { string filename = 1; }

message WorkerReply {
  repeated TempFiles tempFiles = 1;
  string directory = 2;
}

message MasterQuery {
  repeated ShardInfo shard = 1;
  int32 outputNum = 2;
  bool isMap = 3;
  string userId = 4;
  string workerId = 5;
  repeated TempFiles keyFiles = 6;
}

message ShardInfo {
  string filename = 1;
  int32 offStart = 2;
  int32 offEnd = 3;
}

